File: task-3/Python/Sub-Task 4/main.py
import math

fr = open("input.txt")
f = open("output.txt", "w")
n = int(fr.read())
fr.close()
counter = n%2

for i in range(n):
	if counter == 0:
		counter += 2
		continue

	stars = "*"*counter
	spaces = " "*math.floor((n-counter)/2)

	if i < math.floor(n/2):
		counter += 2
	else:
		counter -= 2
	
	f.write(spaces+stars+spaces+"\n")

f.close()

File: task-3/Ruby/Sub-Task 4/main.rb

n = File.read("input.txt").chomp.to_i()
counter = n%2
seq = ""

for i in 0..(n-1)
	if counter == 0
		counter += 2
		next
	end

	stars = "*"*counter
	spaces = " "*(((n-counter)/2).floor())
	
	if i < (n/2).floor()
		counter += 2
	else
		counter -= 2
	end
	
	seq += spaces+stars+spaces+"\n"
end

File.open("output.txt", "w") do |file|
    file.write(seq)
end